### PR TITLE
cleanup tp_parser

### PR DESCRIPTION
**Pull request by satpalsr**
Remove redundant code.

### CODE REVIEW

File: deepspeed/module_inject/auto_tp.py [hunk 1]
```diff
@@ -119,6 +118,5 @@ def tp_parser(model):
                 gem_list = list(set(gem_list))
                 policy_list = AutoTP.update_policy_list(policy_list, module, gem_list)
                 gem_list = []
-        assert len(policy_list), "AutoTP not supported for model. Please use kernel injection since container policy for model exists." \
-        if AutoTP.kernel_supported(module_list) else "Not able to determine model policy automatically. Please provide policy."
+        assert len(policy_list), "AutoTP not supported for model. Please use kernel injection since container policy for model exists."
```
Comment (molly-smith) [inline deepspeed/module_inject/auto_tp.py:121]: Not all models that fail here have kernel support.

**Review by molly-smith — CHANGES_REQUESTED** [review]

File: deepspeed/module_inject/auto_tp.py (hunk 1 above)
Comment (satpalsr) [reply deepspeed/module_inject/auto_tp.py:121]: kernel support was checked twice. [Here](https://github.com/microsoft/DeepSpeed/blob/49fb30810127764bcad40540f0d0edf905cfc9fe/deepspeed/module_inject/auto_tp.py#L99) & [here](https://github.com/microsoft/DeepSpeed/blob/f1e4fb0b878e48d6fe33dbb2e0969d791d698e86/deepspeed/module_inject/auto_tp.py#L123).

Comment (satpalsr) [reply deepspeed/module_inject/auto_tp.py]: Hm, correct. Thanks.

**Comment by satpalsr** [comment]
Hey @molly-smith , Look #3130 
I think we should assert len(policy_list) == 2
and also change this
https://github.com/microsoft/DeepSpeed/blob/7faae44d507ea609696094b2fac2300ac3b24562/deepspeed/module_inject/auto_tp.py#L114

to allow more layers?
```
elif any(cond in layer for cond in ['outer', 'attention.dense']):
```

What to you think?

**Comment by molly-smith** [comment]
> Hey @molly-smith , Look #3130 I think we should `assert len(policy_list) == 2` and also change this
> 
> https://github.com/microsoft/DeepSpeed/blob/7faae44d507ea609696094b2fac2300ac3b24562/deepspeed/module_inject/auto_tp.py#L114
> 
> to allow more layers?
> 
> ```
> elif any(cond in layer for cond in ['outer', 'attention.dense']):
> ```
> 
> What do you think?

Looks like this issue was resolved. Closing this PR.

**Pull request closed without merging**